src/CTA.Rules.Update/CodeReplacer.cs

```diff
@@ -161,15 +163,24 @@ private List<GenericActionExecution> AddActionsWithoutExecutions(FileActions cur
 
         public List<GenericActionExecution> ValidateActions(List<GenericActionExecution> actions, string fileResult)
         {
-            var trimmedResult = Utils.EscapeAllWhitespace(fileResult);
+            // Matches all types of comments and strings
+            string regComments = @"\/\*(?:(?!\*\/)(?:.|[\r\n]+))*\*\/|\/\/(.*?)\r?\n|""((\\[^\n]|[^""\n])*)""|@(""[^""""]*"")+";
 
             foreach (var action in actions)
             {
                 var actionValidation = action.ActionValidation;
+                var trimmedResult = fileResult;
                 var actionValid = true;
 
                 if (actionValidation == null) { continue; }
 
+                if (string.IsNullOrEmpty(actionValidation.CheckComments) || !bool.Parse(actionValidation.CheckComments))
```

Comment:
Changing `CheckComments` to a bool type would simplify this `if` statement as well

Reply:
If keeping it as a string, this looks ok to me 👍